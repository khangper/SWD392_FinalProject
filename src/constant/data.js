export const API_KEY = "API_KEY";

export const FETCH_HOME_LIVESTREAM_REQUEST = "FETCH_HOME_LIVESTREAM_REQUEST";
export const FETCH_HOME_LIVESTREAM_SUCCESS = "FETCH_HOME_LIVESTREAM_SUCCESS";
export const FETCH_HOME_LIVESTREAM_FAILURE = "FETCH_HOME_LIVESTREAM_FAILURE";

export const FETCH_HOME_FEATURED_COURSES_REQUEST =
  "FETCH_HOME_FEATURED_COURSES_REQUEST";
export const FETCH_HOME_FEATURED_COURSES_SUCCESS =
  "FETCH_HOME_FEATURED_COURSES_SUCCESS";
export const FETCH_HOME_FEATURED_COURSES_FAILURE =
  "FETCH_HOME_FEATURED_COURSES_FAILURE";
// All Live Stream
export const FETCH_STREAMS_REQUEST = "FETCH_STREAMS_REQUEST";
export const FETCH_STREAMS_SUCCESS = "FETCH_STREAMS_SUCCESS";
export const FETCH_STREAMS_FAILURE = "FETCH_STREAMS_FAILURE";

//All Instructor
export const FETCH_INSTRUCTORS_REQUEST = "FETCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTORS_SUCCESS = "FETCH_INSTRUCTORS_SUCCESS";
export const FETCH_INSTRUCTORS_FAILURE = "FETCH_INSTRUCTORS_FAILURE";
export const SEARCH_INSTRUCTORS_REQUEST = "SEARCH_INSTRUCTORS_REQUEST";
export const FETCH_INSTRUCTOR_BY_ID_REQUEST = "FETCH_INSTRUCTOR_BY_ID_REQUEST";

// SaveCourse.js
export const FETCH_SAVECOURSES_REQUEST = "FETCH_SAVECOURSES_REQUEST";
export const FETCH_SAVECOURSES_SUCCESS = "FETCH_SAVECOURSES_SUCCESS";
export const FETCH_SAVECOURSES_FAILURE = "FETCH_SAVECOURSES_FAILURE";
export const DELETE_SAVECOURSE_REQUEST = "DELETE_SAVECOURSE_REQUEST";
export const DELETE_SAVECOURSE_SUCCESS = "DELETE_SAVECOURSE_SUCCESS";
export const DELETE_SAVECOURSE_FAILURE = "DELETE_SAVECOURSE_FAILURE";

// Shoppingcart.js
export const ShoppingCart_API_URL =
  "https://66908916c0a7969efd9c67ed.mockapi.io/ojt-repo/ShoppingCart";
export const ADD_TO_CART = "ADD_TO_CART";
export const REMOVE_FROM_CART = "REMOVE_FROM_CART";
export const APPLY_DISCOUNT = "APPLY_DISCOUNT";
export const ADD_TO_CART_FAILURE = " ADD_TO_CART_FAILURE";
export const ADD_TO_CART_SUCCESS = "APPLY_DISCOUNT";

//MyCourse_Course
export const FETCH_COURSES_REQUEST = "FETCH_COURSES_REQUEST";
export const FETCH_COURSES_SUCCESS = "FETCH_COURSES_SUCCESS";
export const FETCH_COURSES_FAILURE = "FETCH_COURSES_FAILURE";

export const EDIT_COURSE_REQUEST = "EDIT_COURSE_REQUEST";
export const EDIT_COURSE_SUCCESS = "EDIT_COURSE_SUCCESS";
export const EDIT_COURSE_FAILURE = "EDIT_COURSE_FAILURE";

export const DELETE_COURSE_REQUEST = "DELETE_COURSE_REQUEST";
export const DELETE_COURSE_SUCCESS = "DELETE_COURSE_SUCCESS";
export const DELETE_COURSE_FAILURE = "DELETE_COURSE_FAILURE";

//UpcommingCourse_Course
export const FETCH_UPCOMMINGCOURSES_REQUEST = "FETCH_UPCOMMINGCOURSES_REQUEST";
export const FETCH_UPCOMMINGCOURSES_SUCCESS = "FETCH_UPCOMMINGCOURSES_SUCCESS";
export const FETCH_UPCOMMINGCOURSES_FAILURE = "FETCH_UPCOMMINGCOURSES_FAILURE";

export const EDIT_UPCOMMINGCOURSE_REQUEST = "EDIT_UPCOMMINGCOURSE_REQUEST";
export const EDIT_UPCOMMINGCOURSE_SUCCESS = "EDIT_UPCOMMINGCOURSE_SUCCESS";
export const EDIT_UPCOMMINGCOURSE_FAILURE = "EDIT_UPCOMMINGCOURSE_FAILURE";

export const DELETE_UPCOMMINGCOURSE_REQUEST = "DELETE_UPCOMMINGCOURSE_REQUEST";
export const DELETE_UPCOMMINGCOURSE_SUCCESS = "DELETE_UPCOMMINGCOURSE_SUCCESS";
export const DELETE_UPCOMMINGCOURSE_FAILURE = "DELETE_UPCOMMINGCOURSE_FAILURE";
