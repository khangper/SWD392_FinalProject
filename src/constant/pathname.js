export const PATH_NAME = {
  HOME: "/home",
  EXPLORE: "/explore",
  LIVE_OUTPUT: "/live_output/:id",
  ADD_LIVE_STREAM: "/add_livestream",
  ALL_LIVE_STREAM: "/all_livestream",
  ALL_INSTRUCTOR_LIST: "/allinstructorlist/:id",
  HELP: "/help",
  REPORT_HISTORY: "/reporthistory",
  SEND_FEEDBACK: "/sendfeedback",
  OTHER_INSTRUCTOR_VIEW: "/other-instructor-view/:id",
  MY_INSTRUCTOR_PROFILE: "/my-instructor-profile",
  SAVE_COURSES: "/save-courses",
  SETTING: "/setting",
  MY_COURSE: "/courses",
  LOGIN: "/login",
  FORGOT_PASSWORD: "/forgot-password",
  SIGN_UP: "/sign-up",
  SIGN_UP_STEP: "/SignUpStep",
  THANKS: "/thanks",
  ERROR_404: "/error404",
  INVOICE: "/invoice",
  PRESS: "/press",
  SEARCH_RESULT: "/search_result",
  PAID_MEMBERSHIP: "/paidmembership",
  ABOUT_US: "/about-us",
  OUR_BLOG: "/our-blog",
  BLOG_SINGLE_VIEW: "/blog-single-view",
  CHECKOUT: "/checkout",
  SHOPPING_CART: "/shoppingcart",
  HELP_VIEW_1: "/helpview1",
  HELP_VIEW_2: "/helpview2",
  COMPANY_DETAILS: "/company-details",
  CAREER: "/career",
  APPLY_JOB: "/apply-job",
  CONTACT_US: "/contactus",
  TERMS_OF_USE: "/termsofuse",
  CERTIFICATION_FILL_FORM: "/certificationfillform",
  CERTIFICATION_TEST: "/certificates-test",
  TEST_RESULT: "/test-result",
  MESSAGE: "/messages",
  COURSES_DETAIL_VIEW: "/courses-detail-view/:id",
  PREVIEWS: "/reviews",
  COMING_SOON: "/coming-soon",
  INS_DASHBOARD: "/ins-dashboard",
  PURCHASE_COURSE: "/PurchaseCourse",
  NOTIFICATON: "/notifications",
  DOWLOADCOURSE: "/dowloadcourse",
  CERTIFICATION_CENTER: "/certificationcenter",
  SITEMAP: "/sitemap",
  EARNING: "/earning",
  PAYOUT: "/payout",
  VERTIFICATION: "/vertification",
  MYCERTIFICATIONS: "/mycertifications",
  CREDIT: "/credits",
  STATEMENTIS: "/statementis",
  STATEMENTST: "/statementst",
  ANALYTICS: "/analytics",
  CREATE_NEW_COURSE: "/create-course",
  STD_DASHBOARD: "/student-dashboard",
  NOTIFICATIONSTUDENT: "/notificationsstudent",
};
