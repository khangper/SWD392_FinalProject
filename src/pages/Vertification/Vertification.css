.Vertification_Container {
    width: 100%;
    margin-left: 30px
}

.Vertification_Header {
    flex: 0 0 auto;
    width: 100%;
}

.Vertification_Header_Title {
    font-size: 20px;
    margin-top: 0 !important;
    margin-bottom: 8px !important;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    color: #333;
}

.Vertification_Body {
    display: flex;
    justify-content: center !important;
}

.Vertification_Body_Col {
    flex: 0 0 auto;
    width: 50%;
}

.Vertification_Body_Content {
    text-align: center;
    margin-top: 30px;
}

.Vertification_Body_Content img {
    width: 200px;
}

img,
svg {
    vertical-align: middle;
}

.Vertification_Body_Content h4 {
    font-size: 24px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #333;
    margin-top: 40px !important;
    margin-bottom: 24px !important;
}

.Vertification_Body_Content p {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #686f7a;
    line-height: 26px;
    margin-bottom: 0;
}

.Alert_Vertification {
    background: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    margin-top: 30px;
    display: flex;
}

.Alert_Vertification_Container {
    display: inline-block;
    width: 50%;
    margin-left: -2px;
    border-right: 1px solid #efefef;
    border-bottom: 0;
}

.Alert_Vertification_Container:last-child {
    border-right: 0;
}

.Vertification_required_group {
    display: flex;
    padding: 20px;
    align-items: center;
}

.Vertification_edututs_required span {
    display: block;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    margin-left: 10px;
    margin-bottom: 8px;
    text-align: left;
}

b,
strong {
    font-weight: bolder;
}

.Vertification_edututs_required span:last-child {
    margin-bottom: 0;
}

.Vertification_apply_verify_text {
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #686f7a;
    line-height: 26px;
    margin-top: 20px;
    margin-bottom: 0;
}

.Verification_form {
    background: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    padding: 0 30px 30px;
    margin-top: 50px;
}

.Verification_form h4 {
    font-size: 24px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    text-align: center;
    color: #333;
    margin-top: 40px !important;
    margin-bottom: 24px !important;
}

.Verification_Form_CT {
    font-size: 1em;
    position: relative;
}

.Verti_mt_50 {
    margin-top: 50px;
}

.Verti_lbel25 label {
    font-size: 14px;
    font-weight: 600;
    font-family: 'Roboto', sans-serif;
    margin-bottom: 10px !important;
    color: #333;
    text-align: left;
    display: block;
}

.Verti_Textbox_Fullname {
    position: relative;
    font-weight: 400;
    font-style: normal;
    display: inline-flex;
    color: rgba(0, 0, 0, .87);
}

.Verti_swdh19 {
    font-size: 14px !important;
    width: 100%;
}

.Verti_Textbox_Fullname>input {
    padding-left: 1em !important;
    padding-right: 25em !important;
    height: 40px !important;
    font-size: 14px;
    font-weight: 400;
    font-family: 'Roboto', sans-serif;
}

.Verti_prompt {
    border-radius: 3px !important;
}

.Verti_srch_explore {
    background: #fff !important;
    border: 1px solid #e5e5e5 !important;
    color: #333 !important;
    box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
}

.Verti_mt_30 {
    margin-top: 30px;
}

.Verti_input_group {
    position: relative;
    display: flex;
    flex-wrap: wrap;
    align-items: stretch;
    width: 100%;
}

.Verti_custom_file {
    position: relative;
    flex: 1 1 auto;
    width: 1%;
    min-width: 0;
    margin-bottom: 0;
    background: #f7f7f7;
}

.Verti_custom_file_label {
    position: absolute;
    top: 0;
    right: 0;
    left: 0;
    z-index: 1;
    height: calc(1.5em + .75rem + 2px);
    padding: 1.3rem 1rem;
    font-weight: 500 !important;
    line-height: 0;
    color: #686f7a !important;
    background-color: #fff;
    border: 1px solid #efefef;
    border-radius: 3px;
    margin-bottom: 0;
    box-shadow: inset 0px 1px 4px rgba(0, 0, 0, 0.02) !important;
}

.Verti_custom_file_label_btn {
    position: absolute;
    top: 0;
    right: 0;
    bottom: 0;
    z-index: 3;
    display: block;
    height: calc(1.5em + .75rem);
    padding: 1.3rem 2rem;
    line-height: 0;
    color: #fff;
    content: "Upload File";
    background-color: #ed2a26;
    border-left: inherit;
    border-radius: 0 3px 3px 0;
    border: none;

}

.Verify_submit_btn {
    font-size: 14px;
    font-weight: 500;
    font-family: 'Roboto', sans-serif;
    color: #fff;
    background: #ed2a26;
    padding: 0px 30px;
    border-radius: 3px;
    border: 0;
    height: 40px;
    margin-top: 45px;
    width: 100%;
    cursor: pointer;
}