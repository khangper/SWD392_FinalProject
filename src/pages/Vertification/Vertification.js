import React from 'react'
import './Vertification.css'
const Vertification = () => {
    return (
        <div className='Vertification_Container'>
            <div className='Vertification_Header'>
                <h2 className="Vertification_Header_Title">
                    Verification
                </h2>
            </div>

            <div className='Vertification_Body'>
                <div className='Vertification_Body_Col'>
                    <div className='Vertification_Body_Content'>
                        <img src="https://gambolthemes.net/html-items/cursus-new-demo/images/verified-account.svg" alt="" />
                        <h4>Verification with Edututs+</h4>
                        <p>
                            Praesent sed sapien gravida, tempus nunc nec, euismod turpis. Mauris quis scelerisque arcu. Quisque et aliquet nisl, id placerat est. Morbi quis imperdiet nulla.
                        </p>

                        <div className='Alert_Vertification'>
                            <div className='Alert_Vertification_Container'>
                                <div className="Vertification_required_group">
                                    <div className="Vertification_edututs_required_img">
                                        <i className="uil uil-dashboard"></i>
                                    </div>
                                    <div className="Vertification_edututs_required">
                                        <span><strong>14 subscribers</strong></span>
                                        <span>500 required</span>
                                    </div>
                                </div>
                            </div>

                            <div className='Alert_Vertification_Container'>
                                <div className="Vertification_required_group">
                                    <div className="Vertification_edututs_required_img">
                                        <i className="uil uil-dashboard"></i>
                                    </div>
                                    <div className="Vertification_edututs_required">
                                        <span>
                                            <strong>10 public watch hours</strong>
                                        </span>
                                        <span>1,500 required</span>
                                    </div>
                                </div>
                            </div>
                        </div>

                        <div className="Vertification_apply_verify_text">
                            <i className="uil uil-check-circle"></i>
                            We'll send you an email when you're eligible to apply
                        </div>

                        <div className="Verification_form">
                            <h4>Verify Your ID</h4>
                            <form>
                                <div className="Verification_Form_CT Verti_mt_50 Verti_lbel25">
                                    <label>Course Title*</label>
                                    <div className="Verti_Textbox_Fullname Verti_swdh19">
                                        <input className="Verti_prompt Verti_srch_explore" type="text" placeholder="Full Name" name="fullname" id="full[name]" defaultValue="" />
                                    </div>
                                </div>
                                <div className="Verification_Form_UD Verti_mt_30 Verti_lbel25">
                                    <label>Upload Document*</label>
                                    <div className="Verti_input_group">
                                        <div className="Verti_custom_file">
                                            <input type="file" className="Verti_custom_file_input" id="inputGroupFile06" />
                                            <label className="Verti_custom_file_label" >
                                                No Choose
                                                <button className='Verti_custom_file_label_btn'>
                                                    Upload File
                                                </button>
                                            </label>
                                        </div>
                                    </div>
                                </div>
                                <button className="Verify_submit_btn" type="submit">Submit Now</button>
                            </form>
                        </div>
                    </div>
                </div>
            </div>
        </div>

    )
}

export default Vertification