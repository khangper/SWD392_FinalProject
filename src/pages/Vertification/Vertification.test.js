import React from "react";
import { renderWithProviders } from "../../constant/test-util";
import Vertification from "../Vertification/Vertification";
import { screen } from "@testing-library/react";

describe("Vertification", () => {
  beforeAll(() => {
    jest.spyOn(console, "error").mockImplementation(() => {});
    jest.spyOn(console, "log").mockImplementation(() => {});
  });

  afterAll(() => {
    console.error.mockRestore();
    console.log.mockRestore();
  });
  it("renders the Vertification component", async () => {
    const { container } = renderWithProviders(<Vertification />);
    expect(container).toBeDefined();
    expect(container).toMatchSnapshot();
  });
});
