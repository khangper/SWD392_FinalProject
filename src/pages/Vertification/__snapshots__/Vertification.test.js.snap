// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Vertification renders the Vertification component 1`] = `
<div>
  <div
    class="Vertification_Container"
  >
    <div
      class="Vertification_Header"
    >
      <h2
        class="Vertification_Header_Title"
      >
        Verification
      </h2>
    </div>
    <div
      class="Vertification_Body"
    >
      <div
        class="Vertification_Body_Col"
      >
        <div
          class="Vertification_Body_Content"
        >
          <img
            alt=""
            src="https://gambolthemes.net/html-items/cursus-new-demo/images/verified-account.svg"
          />
          <h4>
            Verification with Edututs+
          </h4>
          <p>
            Praesent sed sapien gravida, tempus nunc nec, euismod turpis. Mauris quis scelerisque arcu. Quisque et aliquet nisl, id placerat est. Morbi quis imperdiet nulla.
          </p>
          <div
            class="Alert_Vertification"
          >
            <div
              class="Alert_Vertification_Container"
            >
              <div
                class="Vertification_required_group"
              >
                <div
                  class="Vertification_edututs_required_img"
                >
                  <i
                    class="uil uil-dashboard"
                  />
                </div>
                <div
                  class="Vertification_edututs_required"
                >
                  <span>
                    <strong>
                      14 subscribers
                    </strong>
                  </span>
                  <span>
                    500 required
                  </span>
                </div>
              </div>
            </div>
            <div
              class="Alert_Vertification_Container"
            >
              <div
                class="Vertification_required_group"
              >
                <div
                  class="Vertification_edututs_required_img"
                >
                  <i
                    class="uil uil-dashboard"
                  />
                </div>
                <div
                  class="Vertification_edututs_required"
                >
                  <span>
                    <strong>
                      10 public watch hours
                    </strong>
                  </span>
                  <span>
                    1,500 required
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="Vertification_apply_verify_text"
          >
            <i
              class="uil uil-check-circle"
            />
            We'll send you an email when you're eligible to apply
          </div>
          <div
            class="Verification_form"
          >
            <h4>
              Verify Your ID
            </h4>
            <form>
              <div
                class="Verification_Form_CT Verti_mt_50 Verti_lbel25"
              >
                <label>
                  Course Title*
                </label>
                <div
                  class="Verti_Textbox_Fullname Verti_swdh19"
                >
                  <input
                    class="Verti_prompt Verti_srch_explore"
                    id="full[name]"
                    name="fullname"
                    placeholder="Full Name"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                class="Verification_Form_UD Verti_mt_30 Verti_lbel25"
              >
                <label>
                  Upload Document*
                </label>
                <div
                  class="Verti_input_group"
                >
                  <div
                    class="Verti_custom_file"
                  >
                    <input
                      class="Verti_custom_file_input"
                      id="inputGroupFile06"
                      type="file"
                    />
                    <label
                      class="Verti_custom_file_label"
                    >
                      No Choose
                      <button
                        class="Verti_custom_file_label_btn"
                      >
                        Upload File
                      </button>
                    </label>
                  </div>
                </div>
              </div>
              <button
                class="Verify_submit_btn"
                type="submit"
              >
                Submit Now
              </button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
